crates/red_knot_ide/src/goto.rs

```diff
@@ -860,24 +861,20 @@ f(**kwargs<CURSOR>)
                 return "No type definitions found".to_string();
             }
 
-            let mut buf = vec![];
+            let mut buf = String::new();
 
             let source = targets.range;
 
+            let config = DisplayDiagnosticConfig::default()
+                .color(false)
+                .format(DiagnosticFormat::Full);
             for target in &*targets {
-                GotoTypeDefinitionDiagnostic::new(source, target)
-                    .into_diagnostic()
-                    .print(
-                        &self.db,
-                        &DisplayDiagnosticConfig::default()
-                            .color(false)
-                            .format(DiagnosticFormat::Full),
-                        &mut buf,
-                    )
-                    .unwrap();
+                let diag = GotoTypeDefinitionDiagnostic::new(source, target).into_diagnostic();
+                let display = diag.display(&self.db, &config);
+                write!(buf, "{display}").unwrap();
```

Comment:
Nit: In case you only created the `display` variable so that you can used name format args
```suggestion
                write!(buf, "{diag}", diag=diag.display(&self.db, &config)).unwrap();
```

Reply:
It was originally a bigger expression, but sure, yeah, fixed.